api/apiclient.go

```diff
@@ -498,11 +505,14 @@ func dialWebsocket(addr, path string, opts DialOpts, tlsConfig *tls.Config, try
 	return try.Start(newWebsocketDialer(cfg, opts))
 }
 
-// newWebsocketDialer returns a function that
-// can be passed to utils/parallel.Try.Start.
-var newWebsocketDialer = createWebsocketDialer
+// newWebsocketDialer is defined as a variable so that it can be overridden
+// for tests.
+var newWebsocketDialer = newWebsocketDialer0
```

Comment:
To address your comment, @natefinch: "broken windows principle". We must start addressing our technical debt, or risk being crushed by it. I think this is also a decision we've made: moving forward, we fix things as we touch them.

Also, exposing one new function which takes in a dialer would not be a very invasive change and would allow us to remove the patched global variables.

To address your comment, @rogpeppe: I agree, usually exposing these kinds of things reveals a higher-level abstraction which has been missing all along. Probably not worth adressing that in this PR, but it doesn't have to be all or nothing.